anilistUpdater/main.lua

```diff
@@ -244,25 +254,33 @@ function update_anilist(action)
     local cmd = mp.command_native_async(table, callback)
 end
 
-mp.observe_property("percent-pos", "number", check_progress)
+mp.observe_property("pause", "bool", on_pause_change)
 
--- Reset triggered
+-- Reset triggered and start/stop timer based on file loading
 mp.register_event("file-loaded", function()
     triggered = false
-    last_percent_check_time = 0
+
+    -- Stop existing timer if any
+    if progress_timer then
+        progress_timer:kill()
+        progress_timer = nil
+    end
+
     if #DIRECTORIES > 0 then
         local path = get_path()
 
         if not path_starts_with_any(path, DIRECTORIES) then
-            mp.unobserve_property(check_progress)
+            return
         else
             -- If it starts with the directories, check if it starts with any of the excluded directories
             if #EXCLUDED_DIRECTORIES > 0 and path_starts_with_any(path, EXCLUDED_DIRECTORIES) then
-                mp.unobserve_property(check_progress)
+                return
             end
-
         end
     end
+
+    -- Start timer for this file
+    progress_timer = mp.add_periodic_timer(UPDATE_INTERVAL, check_progress)
```

Comment:
The timer should probably only be initialised once outside of any scopes, maybe above the keybinds for example. Theres no reason for setting it to `nil` and then initialising again